service/src/main/java/uk/gov/hmcts/reform/fpl/controllers/NoticeOfProceedingsController.java

```diff
@@ -100,7 +100,8 @@ public AboutToStartOrSubmitCallbackResponse handleAboutToSubmitEvent(
         caseDetails.getData().put("noticeOfProceedings", noticeOfProceedings);
         caseData = mapper.convertValue(caseDetails.getData(), CaseData.class);
 
-        Map<String, Object> templateData = noticeOfProceedingsService.getNoticeOfProceedingTemplateData(caseData);
+        Map<String, Object> templateData = noticeOfProceedingsService
+            .getNoticeOfProceedingTemplateData(caseData).toMap(mapper);
```

Comment:
I think we might want to extract this into its own service and have it extend `DocmosisTemplateDataGeneration`. This way we can always use the `DocmosisNoticeOfProceeding` instead of `Map` to generate the docmosis document.

Reply:
done